src/matlab2tikz.m

```diff
@@ -3543,60 +3543,144 @@ function warnFacetedInterp(m2t)
+            if hasEdgeColor
+                markerOptions = opts_add(markerOptions, 'draw', ecolor);
+            else
+                markerOptions = opts_add(markerOptions, 'draw', xcolor);
+            end
+            if hasFaceColor
+                markerOptions = opts_add(markerOptions, 'fill', xcolor);
+            end
+            % for changing marker size, the 'scatter' option has to be added
+            drawOptions = opts_add(drawOptions, 'color', xcolor);
+            drawOptions = opts_addSubOpts(drawOptions, 'mark options', ...
+                                       markerInfo.options);
+
+            if ~hasFaceColor
+                drawOptions = opts_add(drawOptions, ...
+                    'scatter/use mapped color', xcolor);
+            else
+                drawOptions = opts_addSubOpts(drawOptions, ...
+                    'scatter/use mapped color', markerOptions);
+            end
+        end
+    elseif size(dataInfo.C,2) == 3
+        % scatter plots with each marker a different RGB color (not yet supported!)
+        userWarning(m2t, 'Pgfplots cannot handle RGB scatter plots yet.');
+        % TODO Get this in order as soon as Pgfplots can do "scatter rgb".
+        % See e.g. http://tex.stackexchange.com/questions/197270 and #433
     else
-        [m2t, drawOptions] = getScatterOptsColormap(m2t, h, drawOptions, markerInfo);
+        % scatter plot where the colors are set using a color map
+        markerOptions = opts_new();
+        markerOptions = opts_addSubOpts(markerOptions, 'mark options', ...
+                                     markerInfo.options);
+        if markerInfo.hasEdgeColor && markerInfo.hasFaceColor
+            [m2t, ecolor] = getColor(m2t, h, markerInfo.EdgeColor, 'patch');
+            markerOptions = opts_add(markerOptions, 'draw', ecolor);
+        else
+            markerOptions = opts_add(markerOptions, 'draw', 'mapped color');
+        end
+        if markerInfo.hasFaceColor
+            markerOptions = opts_add(markerOptions, 'fill', 'mapped color');
+        end
+
+        if numel(dataInfo.Size) == 1
+            drawOptions = opts_add(drawOptions, 'mark size', ...
+                sprintf('%.4fpt', dataInfo.Size)); % FIXME: investigate whether to use `m2t.ff` 
+        else
+            %TODO: warn the user about this. It is not currently supported.
+        end
+
+        drawOptions = opts_add(drawOptions, 'scatter src', 'explicit');
+        drawOptions = opts_addSubOpts(drawOptions, 'scatter/use mapped color', ...
+                                   markerOptions);
+        % Add color map.
+    m2t = m2t_addAxisOption(m2t, matlab2pgfplotsColormap(m2t, m2t.current.colormap), []);
     end
     % - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
     % Plot the thing.
-    [env, data, sColumn] = organizeScatterData(m2t, xData, yData, zData, sData);
+    [env, data, metaPart, columns] = organizeScatterData(m2t, dataInfo);
```

Comment:
Im not 100% decided whether we should keep it or just remove it for the 1.1 release as that would be cleaner.